### PR TITLE
feat: adding functionality when creating a new answer

### PR DESCRIPTION
**Pull request by HuziMuzi**
Add and delete new answer also choice correct answer.

image

### CODE REVIEW

File: src/bll/createTestReducer.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export type CreateTestType = {
+  title: string;
+  content: answerType[];
+  type: string;
+  difficulty: difficultyType;
+  description: string;
+  topic: string;
+};
+
+const slice = createSlice({
```
Comment (antoniosk10) [inline src/bll/createTestReducer.ts:19]: testSlice

File: src/bll/createTestReducer.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  name: 'createTest',
```
Comment (antoniosk10) [inline src/bll/createTestReducer.ts:20]: test

File: src/bll/createTestReducer.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  initialState: {
```
Comment (antoniosk10) [inline src/bll/createTestReducer.ts:21]: lets declare initialState above

File: src/components/AddingAnswer/index.tsx [hunk 4]
```diff
@@ -1,16 +1,47 @@
...
+  onPressCorrectAnswer: (id: number, checked: boolean) => void;
+};
+
+export const AddingAnswer = ({
+  item,
+  onPressDelete,
+  onPressCorrectAnswer,
+  disabledDeleteBtn,
+}: AddingAnswerPropsType) => {
+  const onPressDeleteHandler = () => {
```
Comment (antoniosk10) [inline src/components/AddingAnswer/index.tsx:23]: useCallback

File: src/components/AddingAnswer/index.tsx [hunk 5]
```diff
@@ -1,16 +1,47 @@
...
+    onPressDelete(item.id);
+  };
+  const onPressCorrectAnswerHandler = (checked: boolean) => {
```
Comment (antoniosk10) [inline src/components/AddingAnswer/index.tsx:26]: useCallback

**Review by KhaninPasha0488 — APPROVED** [review]

**Review by Julia-Kovalchuk — DISMISSED** [review]

File: src/components/QuestionsTabs/QuestionTab/tests/QuestionTab.test.tsx [hunk 6]
```diff
@@ -3,17 +3,18 @@ import {QuestionTab, QuestionTabPropsType} from '../index';
...
+  id: 1,
   index: 1,
   isActive: true,
   onPress: onPressMock,
 } as QuestionTabPropsType;
 
 describe('Tab component functionality', () => {
   render(<QuestionTab {...createTestProps} />);
-
+  console.log();
```
Comment (antoniosk10) [inline src/components/QuestionsTabs/QuestionTab/tests/QuestionTab.test.tsx:15]: ?

File: src/bll/store/store.ts [hunk 7]
```diff
@@ -1,13 +1,12 @@
 import {configureStore} from '@reduxjs/toolkit';
-import thunkMiddleware from 'redux-thunk';
 import {appReducer} from '../appReducer';
+import {testReducer} from '../testReducer';
 
 export const store = configureStore({
   reducer: {
     app: appReducer,
+    createTest: testReducer,
```
Comment (antoniosk10) [inline src/bll/store/store.ts:8]: rename to testReducer

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  return (
+    <View>
+      <TextBox>Question</TextBox>
+      <BlockBox>
+        <CustomTextInput
+          onChangeText={() => {}}
+          value={question.textQuestion}
+          multiline
+          textAlignVertical={'top'}
+          numberOfLines={Platform.OS === 'ios' ? undefined : 2}
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:55]: you can move this logic outside the component. Platform never change

File: src/screens/CreateTest/QuestionsSettings/index.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      setIdQuestion(id);
+      dispatch(selectCurrentQuestion({id}));
+    },
+    [dispatch],
+  );
+
+  useEffect(() => {
+    dispatch(selectCurrentQuestion({id: idCurrentQuestion}));
+  }, [dispatch, idCurrentQuestion]);
+  return (
```
Comment (antoniosk10) [inline src/screens/CreateTest/QuestionsSettings/index.tsx:29]: add space

File: src/screens/CreateTest/CreateQuestion/CreateAnswer/index.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
+import {View} from 'react-native';
+import {
+  ButtonAnswerBox,
+  TextBox,
+} from '../../../../components/ui/ReadyStyles/Boxes/index';
+import {AddingAnswer} from '../../../../components/AddingAnswer/index';
```
[the remark this answers is not in the document]
Comment (HuziMuzi) [reply src/screens/CreateTest/CreateQuestion/CreateAnswer/index.tsx:6]: Ok

File: src/screens/CreateTest/TestSettings.tsx [hunk 11]
```diff
@@ -32,23 +64,21 @@ export const TestSettings = () => {
...
       </BlockBox>
       <TextBox>Test level</TextBox>
       <BlockBox>
         <SwitchSelectors type="level" />
       </BlockBox>
       <TextBox>Number of questions</TextBox>
       <BlockBox>
-        <SwitchSelectors type="number" />
+        <SwitchSelectors type="number" onPress={selectNumberQuestionsHandler} />
       </BlockBox>
```
Comment (antoniosk10) [inline src/screens/CreateTest/TestSettings.tsx:76]: s

**Review by KhaninPasha0488 — APPROVED** [review]

**Review by Julia-Kovalchuk — APPROVED** [review]

**Pull request merged**